File: Gym/3contest/GreedyFlorist.cpp
#include <iostream>
using namespace std;
void swap(int *xp, int *yp)  
{  
    int temp = *xp;  
    *xp = *yp;  
    *yp = temp;  
}  
  
// A function to implement bubble sort  
void bubbleSort(int arr[], int n)  
{  
    int i, j;  
    for (i = 0; i < n-1; i++)      
      
    // Last i elements are already in place  
    for (j = 0; j < n-i-1; j++)  
        if (arr[j] > arr[j+1])  
            swap(&arr[j], &arr[j+1]);  
}  

int main(){
    int n, k;
    cin >> n >> k;
    int flowers[n];
    for(int i = 0; i < n; ++i){
        cin >> flowers[i];
    }
    bubbleSort(flowers, n);

    int cost, flowersB;
    for(int i = 0; i < n; ++i){
        cost += (1+flowersB) * flowers[n- 1 - i];
        if((i + 1) % k == 0){
            flowersB++;
        }
        
    }
    cout << cost << endl;
}

File: Gym/3contest/GreedyFlorist.py
s = input()
a = s.find(" ")
n = int(s[0:a])
k = int(s[a+1:len(s)])
flowers = [0] * n
s = input() + " "
for i in range(n):
    a = s.find(" ")
    flowers[i] = int(s[0:a])
    s = s[a+1:len(s)]

flowers.sort()

cost = 0
flowersB = 0
for i in range(n):
    cost += (1+flowersB) * flowers[n - 1 - i]
    if((i + 1) % k == 0):
        flowersB += 1

print(cost)
